src/python/pants/engine/process.py

```diff
@@ -48,6 +49,7 @@ def __init__(
         timeout_seconds: Optional[Union[int, float]] = None,
         jdk_home: Optional[str] = None,
         is_nailgunnable: bool = False,
+        execution_slot_variable: str = "",
```

Comment:
Better to make this `Optional[str]`. It expresses intent better. See `jdk_home` as an example.